app/controllers/api_controller/parser/request.rb

```diff
@@ -0,0 +1,81 @@
+class ApiController
+  module Parser
+    class Request
+      def initialize(req, params)
+        @request = req
+        @params = params
+      end
+
+      def action
+        # for basic HTTP POST, default action is "create" with data being the POST body
+        @action ||= method == :put ? 'edit' : (json_body['action'] || 'create')
+      end
+
+      def api_prefix
+        @api_prefix ||= "#{base}#{prefix}"
+      end
+
+      def base
+        url.partition(fullpath)[0] # http://target
+      end
+
+      def collection
+        @params[:collection]
+      end
+
+      def c_id
```

Comment:
It perhaps make sense to rename this also in the `params` hash?

Although, I would like to defer a decision here until I look into compressed ids (https://www.pivotaltracker.com/projects/1610125/stories/121850823) more in depth. We may end-up needing two accessors for each. Don't know for sure yet.